www/js/components/download-track-page.js

```diff
@@ -1,89 +1,108 @@
 import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import { isFinite } from 'underscore'
 
-import { Layout, Header, Content, CardText, ProgressBar, Button } from 'react-mdl';
+import { Switch, Layout, Header, Card, CardActions, CardText, CardTitle, ProgressBar, Button } from 'react-mdl';
 import DeviceStorage from './device-storage';
 import ACDrawer from './ac-drawer';
-import { downloadableTracks } from '../mock-data';
+
+import {
+  fetchTrack,
+  clearTrack,
+  activateTrack,
+  deactivateTrack
+} from '../reducers/tracks'
 
 class DownloadTrackPage extends Component {
   constructor(props) {
     super(props);
-    this.state = {'tracks':downloadableTracks}
   }
 
-  saveTrack(trackId) {
-    // mock saving
-    let modTracks = this.state.tracks;
-    modTracks[trackId].downloading = true;
-    let saving = setInterval(()=>{
-      if (modTracks[trackId].downloaded >= modTracks[trackId].downloadSize || (!modTracks[trackId].downloading)) {
-        modTracks[trackId].downloading = false;
-        this.setState({'tracks': modTracks});
-        clearInterval(saving);
-      } else {
-        modTracks[trackId].downloaded = modTracks[trackId].downloaded + (modTracks[trackId].downloadSize / 300);
-        modTracks[trackId].downloading = true;
-        this.setState({'tracks': modTracks});
-      }
-    }, 50);
+  saveTrack(id, pkg) {
```

Comment:
convention for react function names I believe should be in the style of `onSomethingHappened`, but that's more nit-picky, we can change that later.